x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackSettings.java

```diff
@@ -65,8 +65,10 @@ private XPackSettings() {
     public static final Setting<Boolean> GRAPH_ENABLED = Setting.boolSetting("xpack.graph.enabled", true, Setting.Property.NodeScope);
 
     /** Setting for enabling or disabling machine learning. Defaults to true. */
-    public static final Setting<Boolean> MACHINE_LEARNING_ENABLED = Setting.boolSetting("xpack.ml.enabled", true,
-            Setting.Property.NodeScope);
+    public static final Setting<Boolean> MACHINE_LEARNING_ENABLED = Setting.boolSetting(
```

Comment:
Seems if we are running on arm we want this to _always_ be `false`, not just _default_ to `false`. Wouldn't this mean folks could enable ML explicitly causing funny result?

Reply:
I pushed 7b81079b0e3d5b92eef7cbb0dc100d0362e48e60.